packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -57,18 +58,15 @@ export const useSceneHierarchyData = () => {
   return useContext(Context);
 };
 
-const toSceneHeirarchyNode = ({
-  ref,
-  name,
-  parentRef,
-  childRefs = [],
-  components,
-}: ISceneNodeInternal | Readonly<ISceneNodeInternal>) => {
+const toSceneHeirarchyNode = (
+  { ref, name, parentRef, childRefs = [], components }: ISceneNodeInternal | Readonly<ISceneNodeInternal>,
+  hideChild?: boolean,
```

Comment:
I'm not sure this belongs here, this is kind of polluting the object model. Hiding children in search results is a view concern, it shouldn't affect the underlying data model.

Suggestion: We could move the filtering logic to the getChildNodes hook instead, and just make that aware of the search filters or similar.

Reply:
This might be the simplest way to achieve this though, so it's fine.